### PR TITLE
Add usememos extension

### PR DESCRIPTION
**Pull request by zrr1999**
### Usememos

[Usememos](https://github.com/zrr1999/raycast-memos) is an extension for [usememos/memos](https://github.com/usememos/memos).

#### Preferences

You can customize the following preferences:

- `OpenAI API Key`: API Key is used to authenticate with OpenAI API.
- `Model`: Model is used to specify the model to use.
- `OpenAI Base Path (Optional)`: Base Path is used to override base path with OpenAI API.
- `Language`: LLM will use this language to generate content.
- `Memos Server URL`: The URL of the Memos server.
- `Memos Server Token`: Token is used to authenticate with the memos server.

This PR is auto-generated by [create-pull-request][1]
## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

[1]: https://github.com/peter-evans/create-pull-request

### CODE REVIEW

File: extensions/usememos/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Usememos
+
+[Usememos](https://github.com/zrr1999/usememos) is an extension for [usememos/memos](https://github.com/usememos/memos).
```
Comment (pernielsentikaer) [inline extensions/usememos/README.md:3]: Link is 404

File: extensions/usememos/src/lib/types.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+export interface PreferenceValues {
+  memosServerUrl: string;
+  memosServerToken: string;
+  model: string;
+  language: string;
+  openAiApiKey?: string;
+  openAiBasePath?: string;
+}
+
```
Comment (pernielsentikaer) [inline extensions/usememos/src/lib/types.ts:9]:
```suggestion
```
Already added in raycast-env.d.ts

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- Could you add at least one [screenshot](https://developers.raycast.com/basics/prepare-an-extension-for-store#screenshots) of the extension

- can you record a flow or provide a test key for me :)

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/usememos/README.md (hunk 1 above)
Comment (zrr1999) [reply extensions/usememos/README.md:3]: I made a mistake here, it should be https://github.com/zrr1999/raycast-memos. I will make the correction.

File: extensions/usememos/src/lib/types.ts (hunk 2 above)
Comment (zrr1999) [reply extensions/usememos/src/lib/types.ts:9]: Alright, I just realized that I can use this here. I will make the correction immediately.

**Comment by zrr1999** [comment]
> Hi 👋
> 
> Thanks for your contribution 💪
> 
> I have now tested your extension, and I have some feedback ready for you:
> 
> * Could you add at least one [screenshot](https://developers.raycast.com/basics/prepare-an-extension-for-store#screenshots) of the extension
> * can you record a flow or provide a test key for me :)
> 
> I'm looking forward to testing this extension again 🔥
> 
> Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

I sent the key via a private message on Slack. I will add the video and screenshots shortly.

**Comment by zrr1999** [comment]
Export-1728486299947

File: extensions/usememos/package.json [hunk 3]
```diff
@@ -0,0 +1,102 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "usememos",
+  "title": "Usememos",
+  "description": "Usememos is an extension for usememos/memos",
+  "icon": "icon.png",
+  "owner": "bone-corp",
```
Comment (pernielsentikaer) [inline extensions/usememos/package.json:7]: Can we remove this so it's release under your name instead?

Comment (zrr1999) [reply extensions/usememos/package.json:7]: Of course.

Comment (zrr1999) [inline extensions/usememos/package.json:7]:
```suggestion
```

**Review by zrr1999 — COMMENTED** [review]
rm owner

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Comment by Muzosh** [comment]
@zrr1999 Can you please explain why this extension requires OpenAI token?

**Pull request merged**